crates/topos-tce-broadcast/src/double_echo/broadcast_state.rs

```diff
@@ -65,7 +71,7 @@ impl BroadcastState {
             });
         }
 
-        state.update_status();
+        state.update_status().await;
```

Comment:
I would want to keep the `BroadcastState` non async as it is somehow a state machine.